src/MetadataUtility/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -0,0 +1,165 @@
+        public const string LogFileKey = "Frontier Labs Log file";
+        public const string FirmwareString = "Firmware:";
+        public const string SDCardString = "SD Card :";
+        public const string RecordingString = "New recording started:";
+
+        public LogFile(string filePath)
+        {
+            this.FilePath = filePath;
+        }
+
+        public List<(MemoryCard MemoryCard, int Line)> MemoryCardLogs { get; } = new List<(MemoryCard MemoryCard, int Line)>();
+
+        public List<(string Recording, int Line)> RecordingLogs { get; } = new List<(string Recording, int Line)>();
+
+        public float Firmware { get; set; }
+
+        public static void FindLogFile(TargetInformation information, IEnumerable<string> supportFiles)
+        {
+            IEnumerable<string> logFiles = supportFiles.Where(x => new Regex(@".*logfile.*txt").IsMatch(x));
+
+            string first;
+
+            // If one log file was found, return true and add it to known support files for the target
+            if (logFiles.Length() == 1 && IsLogFile(first = logFiles.First()))
+            {
+                IEnumerable<string> knownSupportFilePaths = TargetInformation.KnownSupportFiles.Select(x => x.FilePath);
+
+                if (knownSupportFilePaths.Contains(first))
+                {
+                    information.TargetSupportFiles.Add(LogFileKey, TargetInformation.KnownSupportFiles[knownSupportFilePaths.ToList().IndexOf(first)]);
+                }
+                else
+                {
+                    LogFile logFile = new LogFile(first);
+                    logFile.ExtractInformation();
+
+                    TargetInformation.KnownSupportFiles.Add(logFile);
+                    information.TargetSupportFiles.Add(LogFileKey, logFile);
+                }
+            }
+            else if (logFiles.Length() > 1)
+            {
+                foreach (string log in logFiles)
+                {
+                    if (IsLogFile(log))
+                    {
+                        LogFile logFile;
+                        IEnumerable<string> knownSupportFilePaths = TargetInformation.KnownSupportFiles.Select(x => x.FilePath);
+
+                        if (knownSupportFilePaths.Contains(log))
+                        {
+                            logFile = (LogFile)TargetInformation.KnownSupportFiles[knownSupportFilePaths.ToList().IndexOf(log)];
+                        }
+                        else
+                        {
+                            logFile = new LogFile(log);
+                            logFile.ExtractInformation();
+                            TargetInformation.KnownSupportFiles.Add(logFile);
+                        }
+
+                        // If more than one log file is found, we must check for the file name in the log file
+                        // Won't apply to later firmware versions
```

Comment:
We do, if an SD card gets filled up, when moving to the next SD slot it will create a new log file. We have a sensor that produced "logfile.txt", "logfile (2).txt" and "logfile (3).txt". Will add this case to the fixtures file.

Reply:
Actually seems this is probably just due to logfiles being collapsed into one directory on our end at some point. The sensor will produce one log file per SD slot, but will by default separate these into different directories.

Regardless, that was why I'm checking if there is one log file (where we can assume it matches the target due to directory structure) or multiple (where I then search each logfile for the specific recording name). I'll leave it up to you whether this is worth keeping or not since most people won't have this issue so long as they keep the original directory structure. If not I can keep it on a separate local copy for our purposes.

Reply:
Okay, this use case seems fair enough. 

But so far we don't have any data where this is demonstrated, so the code written to select the right file will never be tested in our unit tests.

Add another fixture: two files from one sensor, one from each sd card, and two log files (that have the window copy name deduplication suffix in them)..

Reply:
Our WAV files are pretty huge (2GB each). I uploaded empty WAV files with the same names, since that's all we need to match with the log files. That sound okay?